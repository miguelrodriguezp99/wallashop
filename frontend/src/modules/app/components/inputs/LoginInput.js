import React from "react";
import "../../styles/inputs/loginInput.css"

//use effect es 
const LoginInput = ({ name, placeholder, type, value, fn }) => {
    if (fn === undefined) {
        return (
            <>
                <div className="form__group field">
                    <input
                        name={name}
                        type={type} className="form__field"
                        placeholder={placeholder}
                        value={value}
                        required="" />
                    <label htmlFor="name" className="form__label">{placeholder}</label>
                </div>
            </>
        );

    } else
        return (
            <>
                <div className="form__group field">
                    <input
                        name={name}
                        type={type} className="form__field"
                        placeholder={placeholder}
                        value={value}
                        onChange={e => fn(e.target.value)}
                        required="" />
                    <label htmlFor="name" className="form__label">{placeholder}</label>
                </div>
            </>
        );
};

export default LoginInput;
