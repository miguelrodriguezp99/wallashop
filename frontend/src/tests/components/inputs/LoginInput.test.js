import LoginInput from '../../../modules/app/components/inputs/LoginInput';
import renderer from 'react-test-renderer';

describe('LoginInput component', () => {
    it('matches snapshot', () => {
      const mockFn = jest.fn();
  
      const tree = renderer
        .create(<LoginInput name="test" placeholder="Enter text" type="text" value="1" fn={mockFn} />)
        .toJSON();
  
      expect(tree).toMatchSnapshot();
    });
  });