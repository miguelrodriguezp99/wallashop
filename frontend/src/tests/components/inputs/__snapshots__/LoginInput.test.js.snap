// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginInput component matches snapshot 1`] = `
<div
  className="form__group field"
>
  <input
    className="form__field"
    name="test"
    onChange={[Function]}
    placeholder="Enter text"
    required=""
    type="text"
    value="1"
  />
  <label
    className="form__label"
    htmlFor="name"
  >
    Enter text
  </label>
</div>
`;
